yarn.config.cjs

```diff
@@ -0,0 +1,654 @@
+        expectWorkspaceField(workspace, 'version');
+
+        // All non-root packages must have a description that ends in a period.
+        expectWorkspaceDescription(workspace);
+
+        // All non-root packages must have the same set of NPM keywords.
+        expectWorkspaceField(workspace, 'keywords', ['MetaMask', 'Ethereum']);
+
+        // All non-root packages must have a homepage URL that includes its name.
+        expectWorkspaceField(
+          workspace,
+          'homepage',
+          `${repositoryUri}/tree/main/packages/${workspaceBasename}#readme`,
+        );
+
+        // All non-root packages must have a URL for reporting bugs that points
+        // to the Issues page for the repository.
+        expectWorkspaceField(workspace, 'bugs.url', `${repositoryUri}/issues`);
+
+        // All non-root packages must specify a Git repository within the
+        // MetaMask GitHub organization.
+        expectWorkspaceField(workspace, 'repository.type', 'git');
+        expectWorkspaceField(
+          workspace,
+          'repository.url',
+          `${repositoryUri}.git`,
+        );
+
+        // If not specified, the license for all non-root packages must be MIT.
+        if (
+          workspace.manifest.license === null ||
+          workspace.manifest.license === undefined
+        ) {
+          expectWorkspaceField(workspace, 'license', 'MIT');
+        }
+
+        // All non-root packages must not have side effects. (An exception is
+        // made for `@metamask/base-controller`).
+        if (workspace.ident !== '@metamask/base-controller') {
+          expectWorkspaceField(workspace, 'sideEffects', false);
+        }
+
+        // All non-root packages must set up ESM- and CommonJS-compatible
+        // exports correctly.
+        expectCorrectWorkspaceExports(workspace);
+
+        // All non-root packages must have the same "build" script.
+        expectWorkspaceField(
+          workspace,
+          'scripts.build',
+          'tsup --config ../../tsup.config.ts --tsconfig ./tsconfig.build.json --clean',
+        );
+
+        // All non-root packages must have the same "build:docs" script.
+        expectWorkspaceField(workspace, 'scripts.build:docs', 'typedoc');
+
+        if (isPrivate) {
+          // All private, non-root packages must not have a "publish:preview"
+          // script.
+          workspace.unset('scripts.publish:preview');
+        } else {
+          // All non-private, non-root packages must have the same
+          // "publish:preview" script.
+          expectWorkspaceField(
+            workspace,
+            'scripts.publish:preview',
+            'yarn npm publish --tag preview',
+          );
+        }
+
+        // No non-root packages may have a "prepack" script.
+        workspace.unset('scripts.prepack');
+
+        // All non-root package must have a valid "changelog:validate" script.
+        expectCorrectWorkspaceChangelogValidationScript(workspace);
+
```

Comment:
A rule to check `changelog:update` script exist is missing

Reply:
Hmm, you're right. This is something we didn't have before, but I can add it. I've added that here: [d408971](https://github.com/MetaMask/core/pull/4546/commits/d408971400d72377c2c451bfa0d1e6fa85054d5d)